### DISCUSSION TITLE
Docker sftpgo vars to https web ui

### DISCUSSION THREAD

**Discussion by Dante3004** [drakkan/sftpgo#700] [Q&A] [answered]
Hi again,

I have tried several configurations to switch the client to https without success. Here is the podman command I use:

podman run \
    -d --restart=always --pod=sftpgo \
    -v /data/volumes/sftpgo-data:/srv/sftpgo:rw,Z \
    -v /data/volumes/sftpgo-config:/var/lib/sftpgo:rw,Z \
    -e "SFTPGO_HTTPD__BINDINGS__ENABLE_HTTPS=true" \
    -e "SFTPGO_HTTPD__BINDINGS__0__PORT=9443" \
    -e "SFTPGO_HTTPD__CERTIFICATE_FILE=/var/lib/sftpgo/cert.pem" \
    -e "SFTPGO_HTTPD__CERTIFICATE_KEY_FILE=/var/lib/sftpgo/key.pem" \
    --name sftpgo \
    drakkan/sftpgo:latest

the logs says :
```
{"level":"debug","time":"2022-01-28T10:33:00.255","sender":"httpd","message":"initializing HTTP server with config {Bindings:[{Address: Port:9443 EnableWebAdmin:true EnableWebClient:true EnableHTTPS:false ClientAuthType:0 TLSCipherSuites:[] ProxyAllowed:[] HideLoginURL:0 RenderOpenAPI:true WebClientIntegrations:[] allowHeadersFrom:[]} {Address: Port:0 EnableWebAdmin:true EnableWebClient:true EnableHTTPS:true ClientAuthType:0 TLSCipherSuites:[] ProxyAllowed:[] HideLoginURL:0 RenderOpenAPI:true WebClientIntegrations:[] allowHeadersFrom:[]} {Address: Port:0 EnableWebAdmin:true EnableWebClient:true EnableHTTPS:true ClientAuthType:0 TLSCipherSuites:[] ProxyAllowed:[] HideLoginURL:0 RenderOpenAPI:true WebClientIntegrations:[] allowHeadersFrom:[]}] TemplatesPath:/usr/share/sftpgo/templates StaticFilesPath:/usr/share/sftpgo/static BackupsPath:/srv/sftpgo/backups OpenAPIPath:/usr/share/sftpgo/openapi WebRoot: CertificateFile:/var/lib/sftpgo/cert.pem CertificateKeyFile:/var/lib/sftpgo/key.pem CACertificates:[] CARevocationLists:[] SigningPassphrase:[redacted] MaxUploadFileSize:1048576000 Cors:{AllowedOrigins:[] AllowedMethods:[] AllowedHeaders:[] ExposedHeaders:[] AllowCredentials:false Enabled:false MaxAge:0}}"}
{"level":"debug","time":"2022-01-28T10:33:00.269","sender":"httpd","message":"TLS certificate \"/var/lib/sftpgo/cert.pem\" successfully loaded"}
{"level":"info","time":"2022-01-28T10:33:00.269","sender":"sftpd","message":"server listener registered, address: [::]:2022"}
{"level":"info","time":"2022-01-28T10:33:00.269","sender":"httpd","message":"server listener registered, address: [::]:9443 TLS enabled: false"}
```

Thanks in advance for your answers :)

**Comment by drakkan** [comment, marked as answer]
Hi,

please try `SFTPGO_HTTPD__BINDINGS__0__ENABLE_HTTPS=true`

**Reply by Dante3004** [reply]
Fine it work now, thanks for your help. Do you have a list with all env vars possible ?

**Reply by drakkan** [reply]
> Fine it work now, thanks for your help. Do you have a list with all env vars possible ?

Not explictly, each configuration parameter can be set using an env var. The env vars are documented [here](https://github.com/drakkan/sftpgo/blob/main/docs/full-configuration.md#environment-variables) and in the same doc are documented all the configuration parameters. You can also read all the (non deprecated) configuration parameters directly from the default [config file](https://github.com/drakkan/sftpgo/blob/main/sftpgo.json). It should be quite easy to convert the configuration parameter name to the env var name

**Reply by fsdrw08** [reply]
is it possible to pass the config by config file instead of env var in docker?

**Reply by fsdrw08** [reply]
ok, I got it, just mount the json config file to /etc/sftpgo/sftpgo.json
